src/databricks/labs/lsql/dashboards.py

```diff
@@ -99,42 +118,44 @@ def deploy_dashboard(
             )
         return dashboard
 
-    def _format_sql_file(self, sql_query, query_path):
-        with query_path.open("w") as f:
-            try:
-                for statement in sqlglot.parse(sql_query):
-                    # see https://sqlglot.com/sqlglot/generator.html#Generator
-                    pretty = statement.sql(
-                        dialect="databricks",
-                        normalize=True,  # normalize identifiers to lowercase
-                        pretty=True,  # format the produced SQL string
-                        normalize_functions="upper",  # normalize function names to uppercase
-                        max_text_width=80,  # wrap text at 120 characters
-                    )
-                    f.write(f"{pretty};\n")
-            except sqlglot.ParseError:
-                f.write(sql_query)
-
-    def _replace_names(self, node: _DataclassInstance, better_names: dict[str, str]):
-        # walk evely dataclass instance recursively and replace names
-        if isinstance(node, _DataclassInstance):
-            for field in node.__dataclass_fields__.values():
+    def with_better_names(self, dashboard: Dashboard) -> Dashboard:
+        """Replace names with human-readable names."""
+        datasets, better_names = [], {}
+        for dataset in dashboard.datasets:
```

Comment:
The idea of replace names is that you contain all manipulation there - that's why we do recursive traversal. Can you rewrite it as "if isinstance(node, Dataset): ..." and move the logic of this method over there.